src/logger.ts

```diff
@@ -12,41 +12,36 @@ enum LogLevel {
     DEBUG,
 }
 /* eslint-enable */
-type LogLevelStrings = keyof typeof LogLevel;
+
+const LOG_PREFIX = 'RandomWallpaper';
+const SETTINGS = new Settings();
```

Comment:
This is actually a bit problematic.

I began doing the stuff needed for Gnome45+ and:
Because this is at a top level it'll initiate a new settings object directly when other stuff imports this module. And the settings object in turn will rely on the extension object already being created in `extension.ts`.

So there's:
1. The module holds onto a settings object without the extension being enabled which is against the guidelines.
2. This creates an impossible dependency loop. (*)

(*): Explaining a bit more:
1. The extension imports the logger module.
2. The logger module creates a settings object.
1. The settings object relies on the extension path.
3. To get the extension path in Gnome45+ we query the extension object.
4. But the extension object isn't constructed at this time because the logger module is loaded first.

Reply:
Hm, that's a good catch ... Now I initialize the settings module lazily. Still, I think this doesn't eliminate the recursive dependency when the logger is used inside the extension module before the extension module is fully constructed.

Do you think this is sufficient for now or do you see another issue with that approach?

Also changed the `getEnum` to `getInt` in `_selectedLogLevel()` so it actually works :sweat_smile:

Reply:
Oh wait, we use it to log "Enable extension." in the extension module :facepalm: ... does it still work for Gnome 45 (I'm still on 44)?

Reply:
Yes it's fine this way.

~~But I think we need a way to let got of the settings object when disabling the extension. Similar to the timer singleton. Logger.dispose() or something?~~
While above is still true, I think we're going to remove it anyway - so don't bother for now.

FYI: Re-using the modules across the shell and preference window is somewhat annoying: [`400c4c9` (#176)](https://github.com/ifl0w/RandomWallpaperGnome3/pull/176/commits/400c4c9a28d41715a7c34a2ee626c88771900f4e)

Because of this, I think I'll remove the creation of the settings object in the logger class altogether and add the settings object as a parameter instead. The settings object should already be available in most places we call the logger. Will have a look at that later.

Reply:
Just added a `destroy` function that clears the settings reference, but I really hope this isn't really necessary.

Reply:
Passing the settings object to some setup function as an argument to some initialize function is fine! Feel free to add that to your PR.

Reply:
But please avoid passing the extension object to every log call. If there is no other option we should rethink the log helper. 

Sidenote, already really dislike that I pass the calling context object to log functions:sweat_smile: .